iceberg/src/test/scala/org/apache/spark/sql/delta/serverSidePlanning/SparkToIcebergExpressionConverterSuite.scala

```diff
@@ -0,0 +1,177 @@
+/*
+ * Copyright (2025) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.delta.serverSidePlanning
+
+import org.apache.spark.sql.sources._
+import org.scalatest.funsuite.AnyFunSuite
+
+class SparkToIcebergExpressionConverterSuite extends AnyFunSuite {
+
+  // Helper: Assert filter converts successfully and output contains expected terms
+  private def assertConverts(filter: Filter, expectedTerms: String*): Unit = {
+    val result = SparkToIcebergExpressionConverter.convert(filter)
+    assert(result.isDefined, s"Should convert: $filter")
+    val expr = result.get.toString
+    expectedTerms.foreach(term =>
+      assert(expr.contains(term), s"Missing '$term' in expression: $expr")
+    )
```

Comment:
if you are comparing term based on the string representation of the icebrg expression.. then how will you assert whether the type of the converted expression is correct? for example if `i < 5` (i.e. compare as integer) is incorrectly converted to iceberg expr `i < "5"` (i.e., compare as string), then how will this check catch that?